### DISCUSSION TITLE
`dump-json`

### DISCUSSION THREAD

**Discussion by amitu** [fastn-stack/fastn#721] [Ideas & RFCs]
`fpm dump-json <file.ftd> <name>` would dump json for a given variable.

If the `<name>` is a `record` or `or-type` it will dump json schema for them.

# Example

Say we have `john.ftd` file:

```ftd
-- record person:
caption name:
body address:

-- person john-doe: John Doe

12 ABC Street, 
DK, 
FictionVille

-- foo:

-- foo: hello
```

`foo.ftd`

```ftd
-- import: john
-- john.john-doe

-- 
```

We can run: `fpm dump-json /john/ john-doe` and it prints the following to stdout:

```json
{
    "name": "John Doe",
    "address": "12 ABC Street,\nDK,\nFictionVille"
}
```

# Semantics

You are asking the first argument, what is the value of second argument.

**Comment by Arpita-Jaiswal** [comment]
How are we passing JSON? The argument only accepts file-name and variable-name. @amitu

**Reply by amitu** [reply]
It will not take json. It will dump variable defined in ftd file as json.
